br/pkg/backup/client.go

```diff
@@ -783,68 +768,24 @@ func WriteBackupDDLJobs(metaWriter *metautil.MetaWriter, g glue.Glue, store kv.S
 	return nil
 }
 
-func (bc *Client) getProgressRanges(ranges []rtree.Range) []*rtree.ProgressRange {
+func (bc *Client) GetProgressRanges(ranges []rtree.Range) []*rtree.ProgressRange {
 	prs := make([]*rtree.ProgressRange, 0, len(ranges))
 	for _, r := range ranges {
 		prs = append(prs, bc.getProgressRange(r))
 	}
 	return prs
 }
 
-func buildProgressRangeTree(pranges []*rtree.ProgressRange) (rtree.ProgressRangeTree, []*kvrpcpb.KeyRange, error) {
+func BuildProgressRangeTree(pranges []*rtree.ProgressRange) (rtree.ProgressRangeTree, error) {
 	// the response from TiKV only contains the region's key, so use the
 	// progress range tree to quickly seek the region's corresponding progress range.
 	progressRangeTree := rtree.NewProgressRangeTree()
-	subRangesCount := 0
 	for _, pr := range pranges {
 		if err := progressRangeTree.Insert(pr); err != nil {
-			return progressRangeTree, nil, errors.Trace(err)
+			return progressRangeTree, errors.Trace(err)
 		}
-		subRangesCount += len(pr.Incomplete)
 	}
-	// either the `incomplete` is origin range itself,
-	// or the `incomplete` is sub-ranges split by checkpoint of origin range.
-	subRanges := make([]*kvrpcpb.KeyRange, 0, subRangesCount)
-	progressRangeTree.Ascend(func(pr *rtree.ProgressRange) bool {
-		for _, r := range pr.Incomplete {
-			subRanges = append(subRanges, &kvrpcpb.KeyRange{
-				StartKey: r.StartKey,
-				EndKey:   r.EndKey,
-			})
-		}
-		return true
-	})
-
-	return progressRangeTree, subRanges, nil
-}
-
-func (bc *Client) getBackupTargetStores(
-	ctx context.Context,
-	replicaReadLabel map[string]string,
-) ([]*metapb.Store, map[uint64]struct{}, error) {
-	allStores, err := conn.GetAllTiKVStoresWithRetry(ctx, bc.mgr.GetPDClient(), connutil.SkipTiFlash)
-	if err != nil {
-		return nil, nil, errors.Trace(err)
-	}
-	var targetStores []*metapb.Store
-	targetStoreIds := make(map[uint64]struct{})
-	if len(replicaReadLabel) == 0 {
-		targetStores = allStores
-	} else {
-		for _, store := range allStores {
-			for _, label := range store.Labels {
-				if val, ok := replicaReadLabel[label.Key]; ok && val == label.Value {
-					targetStores = append(targetStores, store) // send backup push down request to stores that match replica read label
-					targetStoreIds[store.GetId()] = struct{}{} // record store id for fine grained backup
-				}
-			}
-		}
-		if len(targetStores) == 0 {
-			return nil, nil, errors.Errorf("no store matches replica read label: %v", replicaReadLabel)
-		}
-	}
-
-	return targetStores, targetStoreIds, nil
+	return progressRangeTree, nil
```

Comment:
In the PR #52535, it split the function `BuildProgressRangeTree` into `buildProgressRangeTree` and `getProgressRanges` in order to avoid return error in `for + workerpool.Apply`.
So this PR can merge these two functions back,
```
func BuildProgressRangeTree(ranges []rtree.Range) []*rtree.ProgressRange {
...
  for _, r := range ranges {
    pr := bc.getProgressRange(r)
    if err := progressRangeTree.Insert(pr); err != nil {
      return progressRangeTree, errors.Trace(err)
    }
  }
...
```

Reply:
nailed it